bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareBridgeHandler.java

```diff
@@ -0,0 +1,239 @@
+            if (topologyPollingJob != null) {
+                noJob = topologyPollingJob.isCancelled();
+            }
+            if (noJob) {
+                topologyPollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                    petcareAPI.updateTopologyCache();
+                    updateThings();
+                }, config.getRefreshIntervalTopology(), config.getRefreshIntervalTopology(), TimeUnit.SECONDS);
+                logger.debug("Bridge topology polling job every {} seconds", config.getRefreshIntervalTopology());
+            }
+        } else {
+            logger.warn("Invalid setting for topology refresh interval");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-invalid-refresh-intervals");
+        }
+        if (config.getRefreshIntervalStatus() != null) {
+            boolean noJob = true;
+            if (petStatusPollingJob != null) {
+                noJob = petStatusPollingJob.isCancelled();
+            }
+            if (noJob) {
+                petStatusPollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                    pollAndUpdatePetStatus();
+                }, config.getRefreshIntervalStatus(), config.getRefreshIntervalStatus(), TimeUnit.SECONDS);
+                logger.debug("Pet status polling job every {} seconds", config.getRefreshIntervalStatus());
+            }
+        } else {
+            logger.warn("Invalid setting for status refresh interval");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-invalid-refresh-intervals");
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(SurePetcareDiscoveryService.class);
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void dispose() {
+        updateState(BRIDGE_CHANNEL_ONLINE, OnOffType.OFF);
+
+        if (topologyPollingJob != null && !topologyPollingJob.isCancelled()) {
+            topologyPollingJob.cancel(true);
+            topologyPollingJob = null;
+            logger.debug("Stopped pet background polling process");
+        }
+        if (petStatusPollingJob != null && !petStatusPollingJob.isCancelled()) {
+            petStatusPollingJob.cancel(true);
+            petStatusPollingJob = null;
+            logger.debug("Stopped pet location background polling process");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateState(BRIDGE_CHANNEL_ONLINE, OnOffType.from(petcareAPI.isOnline()));
+            updateState(BRIDGE_CHANNEL_REFRESH, OnOffType.OFF);
+        } else {
+            switch (channelUID.getId()) {
+                case BRIDGE_CHANNEL_REFRESH:
+                    if ("ON".equals(command.toString())) {
```

Comment:
It seems you are comparing with the `OnOffType.ON.toString()`better compare with `OnOffType.ON`

Reply:
done